fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.plans.JoinType;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.physical.PhysicalHashJoin;
+import org.apache.doris.nereids.trees.plans.physical.PhysicalOlapScan;
+import org.apache.doris.nereids.trees.plans.physical.PhysicalPlan;
+import org.apache.doris.nereids.trees.plans.physical.RuntimeFilter;
+import org.apache.doris.planner.HashJoinNode;
+import org.apache.doris.planner.HashJoinNode.DistributionMode;
+import org.apache.doris.planner.OlapScanNode;
+import org.apache.doris.planner.RuntimeFilterGenerator.FilterSizeLimits;
+import org.apache.doris.planner.RuntimeFilterId;
+import org.apache.doris.qe.SessionVariable;
+import org.apache.doris.thrift.TRuntimeFilterType;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+/**
+ * generate runtime filter
+ */
+public class RuntimeFilterGenerator extends PlanPostprocessor {
+
+    private static final IdGenerator<RuntimeFilterId> GENERATOR = RuntimeFilterId.createGenerator();
+
+    private final Map<ExprId, List<RuntimeFilter>> filtersByExprId = Maps.newHashMap();
+
+    private final Map<ExprId, List<RuntimeFilter.RuntimeFilterTarget>> filterTargetByTid = Maps.newHashMap();
+
+    private final List<org.apache.doris.planner.RuntimeFilter> origFilters = Lists.newArrayList();
+
+    private final SessionVariable sessionVariable;
+
+    private final FilterSizeLimits limits;
+
+    public RuntimeFilterGenerator(SessionVariable sessionVariable) {
+        this.sessionVariable = sessionVariable;
+        this.limits = new FilterSizeLimits(sessionVariable);
+    }
+
+    @Override
+    public PhysicalPlan visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> join, CascadesContext ctx) {
+        Plan left = join.left();
+        Plan right = join.right();
+        if (join.getJoinType() == JoinType.INNER_JOIN) {
+            List<EqualTo> eqPreds = join.getHashJoinConjuncts().stream()
+                    .map(EqualTo.class::cast).collect(Collectors.toList());
+            List<TRuntimeFilterType> legalTypes = Arrays.stream(TRuntimeFilterType.values()).filter(type ->
+                    (type.getValue() & sessionVariable.getRuntimeFilterType()) > 0).collect(Collectors.toList());
+            List<RuntimeFilter> runtimeFilters = Lists.newArrayList();
+            AtomicInteger cnt = new AtomicInteger();
+            final PhysicalHashJoin<Plan, Plan> joinReplica = join;
```

Comment:
why create this variable

Reply:
for the exprOrder for OriginRuntimeFilter

Reply:
> why create this variable

for the exprOrder for OriginRuntimeFilter